SCons/Tool/applelink.py

```diff
@@ -71,12 +69,7 @@ def _applelib_versioned_lib_soname(env, libnode, version, prefix, suffix, name_f
         print("_applelib_versioned_lib_soname: name={!r}".format(name))
     major = version.split('.')[0]
     (libname,_suffix) = name.split('.')
-    # if a desired SONAME was supplied, use that, otherwise create 
-    # a default from the major version
-    if env.get('SONAME'):
-        soname = ShLibSonameGenerator(env, libnode)
-    else:
-        soname = '.'.join([libname, major, _suffix])
+    soname = '.'.join([libname, major, _suffix])
```

Comment:
Why remove using supplied SONAME?

Reply:
This was not the correct order of the calling functions. ShLibSonameGenerator calls into LINKCALLBACK, which calls this function if SONAME is not supplied. So essentially is should handling SONAME and SOVERSION in ShLibSonameGenerator.